File: problems/1128_number_of_equivalent_domino_pairs/solution.py
class Solution:
    def numEquivDominoPairs(self, dominoes: list[list[int]]) -> int:
        count = {}
        result = 0
        for a, b in dominoes:
            key = (min(a, b), max(a, b))
            result += count.get(key, 0)
            count[key] = count.get(key, 0) + 1
        return result

File: problems/1128_number_of_equivalent_domino_pairs/Solution.java
import java.util.*;

class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        Map<Integer, Integer> count = new HashMap<>();
        int result = 0;
        for (int[] d : dominoes) {
            int key = Math.min(d[0], d[1]) * 10 + Math.max(d[0], d[1]);
            int c = count.getOrDefault(key, 0);
            result += c;
            count.put(key, c + 1);
        }
        return result;
    }
}
